### PR TITLE
feat: add DataUtilTest

### PR DESCRIPTION
**Pull request by KarpovIvan**
## Description

[task: 24657](https://github.com/camunda/camunda/issues/24657) 

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.HashSet;
+import org.junit.jupiter.api.Test;
+
+public class DataUtilTest {
+
+  @Test
+  public void shouldNewHashSetWithMultipleElements() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java:19]:
```suggestion
  public void shouldReturnHashSetWithMultipleElements() {
```

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    // when
+    final HashSet<String> result = DataUtil.newHashSet("apple", "banana", "cherry");
+
+    // then
+    assertThat(result).isNotNull().hasSize(3).contains("apple", "banana", "cherry");
+  }
+
+  @Test
+  public void shouldNewHashSetWithDuplicateElements() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java:28]:
```suggestion
  public void shouldDeduplicateDuplicateElements() {
```

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    // when
+    final HashSet<String> result =
+        DataUtil.newHashSet("apple", "banana", "apple", "cherry", "banana");
+
+    // then
+    assertThat(result).isNotNull().hasSize(3).contains("apple", "banana", "cherry");
+  }
+
+  @Test
+  public void shouldNewHashSetWithEmptyArray() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java:38]:
```suggestion
  public void shouldReturnEmptyHashSet() {
```

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    // when
+    final HashSet<String> result = DataUtil.newHashSet();
+
+    // then
+    assertThat(result).isNotNull().isEmpty();
+  }
+
+  @Test
+  public void shouldNewHashSetWithSingleElement() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java:47]:
```suggestion
  public void shouldReturnHashSetWithSingleElement() {
```

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    // when
+    final HashSet<String> result = DataUtil.newHashSet("apple");
+
+    // then
+    assertThat(result).isNotNull().hasSize(1).contains("apple");
+  }
+
+  @Test
+  public void shouldNewHashSetWithNullArray() {
```
Comment (RomanJRW) [inline optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java:56]:
```suggestion
  public void shouldThrowExceptionForNullArray() {
```

**Review by RomanJRW — APPROVED** [review]
Generally LGTM, just some naming improvements

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java (hunk 1 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java (hunk 2 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java (hunk 3 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java (hunk 4 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java]: fixed

File: optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java (hunk 5 above)
Comment (KarpovIvan) [reply optimize/backend/src/test/java/io/camunda/optimize/service/util/DataUtilTest.java]: fixed

**Pull request merged**